### PR TITLE
Issue #14: Implement source-map schema, validation, and crawl config

### PR DESCRIPTION
**Pull request by stranske**
> **Source:** Issue #14

#### Scope
Coverage quality depends on explicit source-map definitions. We need a validated, auditable source-map layer so discovery runs are reproducible and resilient to site heterogeneity.

### Related Issues/PRs
- [#12](https://github.com/stranske/Pension-Data/issues/12)

#### Tasks
- [x] Define source-map schema (seed URLs, allowed domains, doc-type hints, pagination hints, crawl constraints)
- [x] Implement URL normalization and duplicate detection rules
- [x] Implement schema validator with actionable error reporting
- [x] Support per-system override fields for known website quirks
- [ ] Add source-map linter/QA command for CI usage
- [x] Add tests for malformed maps, duplicate seeds, invalid domains, and override behavior

#### Acceptance criteria
- [x] Source maps fail validation for malformed or ambiguous entries
- [x] Duplicate and conflicting URL entries are detected and reported
- [x] Per-system overrides are supported and test-covered
- [ ] CI can run source-map lint/validate checks deterministically

### CODE REVIEW

File: src/pension_data/sources/lint.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        help="Path to source-map CSV file.",
+    )
+    args = parser.parse_args(argv)
+
+    entries = load_source_map(Path(args.source_map))
+    findings = validate_source_map(entries)
+    if findings:
+        for finding in findings:
+            print(f"[{finding.code}] {finding.plan_id}: {finding.message}")
+        return 1
```
[the remark this answers is not in the document]
Comment (stranske) [reply src/pension_data/sources/lint.py]: Addressed in commit 3a53fa2 with code and test updates.

File: src/pension_data/sources/schema.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    plan_name: str
+    expected_plan_identity: str
+    seed_url: str
+    allowed_domains: tuple[str, ...]
+    doc_type_hints: tuple[str, ...]
+    pagination_hints: tuple[str, ...]
+    crawl_constraints: CrawlConstraints
+    source_authority_tier: SourceAuthorityTier
+    mismatch_reason: MismatchReason | None = None
+    overrides: dict[str, str] | None = None
```
[the remark this answers is not in the document]
Comment (stranske) [reply src/pension_data/sources/schema.py]: Addressed in commit 3a53fa2 with code and test updates.

File: src/pension_data/sources/validate.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+                plan_id=_cell(row, "plan_id"),
+                plan_name=_cell(row, "plan_name"),
+                expected_plan_identity=_cell(row, "expected_plan_identity"),
+                seed_url=_cell(row, "seed_url"),
+                allowed_domains=_split_list(_cell(row, "allowed_domains")),
+                doc_type_hints=_split_list(_cell(row, "doc_type_hints")),
+                pagination_hints=_split_list(_cell(row, "pagination_hints")),
+                crawl_constraints=CrawlConstraints(
+                    max_pages=int(_cell(row, "max_pages")),
+                    max_depth=int(_cell(row, "max_depth")),
```
[the remark this answers is not in the document]
Comment (stranske) [reply src/pension_data/sources/validate.py]: Addressed in commit 3a53fa2 with code and test updates.
